tensorboard/webapp/plugins/npmi/store/npmi_reducers_test.ts

```diff
@@ -268,4 +268,485 @@ describe('npmi_reducers', () => {
+              min: -1.0,
+              includeNaN: true,
+            },
+            'nPMI@second': {
+              max: 1.0,
+              min: -1.0,
+              includeNaN: false,
+            },
+            'nPMI@third': {
+              max: 1.0,
+              min: -1.0,
+              includeNaN: false,
+            },
+          },
+          metricArithmetic: [
+            {kind: 'metric', metric: 'nPMI@test'},
+            {kind: 'operator', operator: Operator.AND},
+            {kind: 'metric', metric: 'nPMI@second'},
+            {kind: 'operator', operator: Operator.AND},
+            {kind: 'metric', metric: 'nPMI@third'},
+          ],
+        });
+        const nextState = reducers(
+          state,
+          actions.removeMetricFilter({metric: 'nPMI@test'})
+        );
+        expect(nextState.metricFilters).toEqual({
+          'nPMI@second': {
+            max: 1.0,
+            min: -1.0,
+            includeNaN: false,
+          },
+          'nPMI@third': {
+            max: 1.0,
+            min: -1.0,
+            includeNaN: false,
+          },
+        });
+        expect(nextState.metricArithmetic).toEqual([
+          {kind: 'metric', metric: 'nPMI@second'},
+          {kind: 'operator', operator: Operator.AND},
+          {kind: 'metric', metric: 'nPMI@third'},
+        ]);
+      });
+
+      it('removing a metric filter in the middle', () => {
+        const state = createNpmiState({
+          metricFilters: {
+            'nPMI@test': {
+              max: 0.3,
+              min: -1.0,
+              includeNaN: true,
+            },
+            'nPMI@second': {
+              max: 1.0,
+              min: -1.0,
+              includeNaN: false,
+            },
+            'nPMI@third': {
+              max: 1.0,
+              min: -1.0,
+              includeNaN: false,
+            },
+          },
+          metricArithmetic: [
+            {kind: 'metric', metric: 'nPMI@test'},
+            {kind: 'operator', operator: Operator.AND},
+            {kind: 'metric', metric: 'nPMI@second'},
+            {kind: 'operator', operator: Operator.AND},
+            {kind: 'metric', metric: 'nPMI@third'},
+          ],
+        });
+        const nextState = reducers(
+          state,
+          actions.removeMetricFilter({metric: 'nPMI@second'})
+        );
+        expect(nextState.metricFilters).toEqual({
+          'nPMI@test': {
+            max: 0.3,
+            min: -1.0,
+            includeNaN: true,
+          },
+          'nPMI@third': {
+            max: 1.0,
+            min: -1.0,
+            includeNaN: false,
+          },
+        });
+        expect(nextState.metricArithmetic).toEqual([
+          {kind: 'metric', metric: 'nPMI@test'},
+          {kind: 'operator', operator: Operator.AND},
+          {kind: 'metric', metric: 'nPMI@third'},
+        ]);
+      });
+    });
+
+    describe('Change a Filter', () => {
+      it('change a metric filter', () => {
```

Comment:
add a test that adds a new metric filter.

Reply:
I think I have that. See lines 440-492.

Reply:
Hmm, technically, though, I can pass in a new metric in the changeMetricFilter and the reducer will gladly add a new entry. Can we put an invariant check in the changeMetricFilter reducer? (i.e., the metric should exist in order to change)